app/services/concerns/xccdf/rule_group_rules.rb

```diff
@@ -0,0 +1,111 @@
+# frozen_string_literal: true
+
+module Xccdf
+  # Methods related to saving RuleGroupRules
+  module RuleGroupRules
+    extend ActiveSupport::Concern
+
+    included do
+      def save_rule_group_rules
+        @op_rules_and_rule_groups = @op_rule_groups + @op_rules
+        @rule_group_rules = rgr_with_requires + rgr_with_conflicts
+        @rule_group_rules = @rule_group_rules.compact
+
+        ::RuleGroupRule.import!(
+          @rule_group_rules.select(&:new_record?), ignore: true
+        )
+
+        rules_base = ::RuleGroupRule.joins(included_rule: :benchmark)
+        rule_groups_base = ::RuleGroupRule.joins(included_rule_group: :benchmark)
+
+        rgr_links_to_remove(rules_base).delete_all
+        rgr_links_to_remove(rule_groups_base).delete_all
+      end
+
+      def save_rule_parents
+        @rule_group_rules = rules_with_rule_group_parent&.compact
+        ::RuleGroupRule.import!(@rule_group_rules.select(&:new_record?),
+                                on_duplicate_key_update: {
+                                  conflict_target: %i[rule_id],
+                                  columns: %i[rule_group_id]
+                                })
+
+        rule_parents_base = ::RuleGroupRule.joins(rule: :benchmark)
+
+        rule_parent_links_to_remove(rule_parents_base).delete_all
+      end
+
+      private
+
+      def rgr_with_requires
+        @rgr_with_requires ||= @op_rules_and_rule_groups.flat_map do |op_r_or_rg|
+          including = rule_or_rule_group_for(ref_id: op_r_or_rg.id)
+          op_r_or_rg.requires.map do |required_ref_id|
+            requiring = rule_or_rule_group_for(ref_id: required_ref_id)
+            next unless requiring
```

Comment:
@skateman Should I make `rgr_with_requires`, `rgr_with_conflicts` and `rules_with_rule_group_parent` public methods? 
If I leave them as private methods, do you know of a good way to test the methods? Codecov wants me to test them but I can't figure out a good way to do that since they're private..